pytorch_lightning/core/lightning.py

```diff
@@ -1562,6 +1572,9 @@ def optimizer_step(
         """
         optimizer.step(closure=optimizer_closure)
 
+        if _TORCH_GREATER_EQUAL_1_10 and self._model_averager is not None:
+            self._model_averager.average_parameters(self.model.parameters())
+
```

Comment:
in this case, what happens if the LightningModule contains multiple modules for training?

Reply:
Wouldn't 'self.model.parameters()' here recursively contain all the child modules? Or are you suggesting that we should still read parameters from `self.optimizers()`?

Reply:
Updated similar to `PostLocalSGDOptimizer`, which only average parameters conditionally;
https://github.com/pytorch/pytorch/blob/master/torch/distributed/optim/post_localSGD_optimizer.py?l#L81-L85